src/silx/io/dictdumplinks/_vds_types.py

```diff
@@ -0,0 +1,84 @@
+from collections.abc import Sequence
+from typing import TypedDict, Any, Literal, NamedTuple
+
+from numpy.typing import DTypeLike
+
+from ._base_types import DsetIndex
+from ._base_types import RawDsetIndex
+from ._base_types import DsetIndexItem
+from ._base_types import RawDsetIndexItem
+
+
+class VdsSourceSchemaV1(TypedDict):
+    file_path: str
+    data_path: str
+    shape: Sequence[int]
+    dtype: str | DTypeLike
+    source_index: DsetIndex
+    target_index: DsetIndex
+
+
+class VdsSchemaV1(TypedDict):
+    dictdump_schema: Literal["external_virtual_link_v1"]
+    shape: Sequence[int]
+    dtype: str | DTypeLike
+    sources: Sequence[VdsSourceSchemaV1]
+
+
+class VdsSource(NamedTuple):
+    file_path: str
+    data_path: str
+    shape: tuple[int]
+    dtype: str | DTypeLike
+    source_index: RawDsetIndex
+    target_index: RawDsetIndex
+
+
+class Vds(NamedTuple):
+    shape: tuple[int]
+    dtype: str | DTypeLike
+    sources: list[VdsSource]
+
+
+def parse_vds_schema_v1(target: VdsSchemaV1) -> Vds:
```

Comment:
@t20100 @payno  I'm using `TypedDict` and `NamedTuple` but I'm basically re-inventing `pydantic`. Can we add  `pydantic` as a base dependency to silx? It has the following dependencies

```
annotated-types-0.7.0 pydantic-2.11.3 pydantic-core-2.33.1 typing-extensions-4.13.2 typing-inspection-0.4.0
```